api/lib/resolvers.js

```diff
@@ -236,11 +237,33 @@ const resolvers = (networkList, notificationController) => ({
         validator
       )
     },
-    profile: (validator, _, { dataSources }) => {
-      return localStore(dataSources, validator.networkId).validators
-        ? localStore(dataSources, validator.networkId).validators[
-            validator.operatorAddress
-          ].profile
+    profile: async (validator, _, { dataSources }) => {
+      const validators = localStore(dataSources, validator.networkId).validators
+      const network = networkList.find(({ id }) => id === validator.networkId)
+      const dataSourceClass = require(`./${network.source_class_name}`)
+      const fiatValuesAPI = new FiatValuesAPI()
+      const networkSchemaName = validator.networkId.replace(/-/g, '_')
+      const db = new database(config)(networkSchemaName)
+      const dataSource = new dataSourceClass(
+        network,
+        localStore(dataSources, validator.networkId),
+        fiatValuesAPI,
+        db
+      )
+      if (validators && !validators[validator.operatorAddress].profile) {
```

Comment:
this will never update the validator profiles as it adds to the store once and then never runs again